pyroomacoustics/beamforming.py

```diff
@@ -722,47 +742,42 @@ def udr(self, source, interferer, f, R_n=None, dB=False):
     def process(self, FD=False):
 
         if self.signals is None or len(self.signals) == 0:
-            raise NameError('No signal to beamform')
+            raise NameError('No signal to beamform.')
 
         if FD is True:
 
             # STFT processing
-
             if self.weights is None and self.filters is not None:
                 self.weights_from_filters()
             elif self.weights is None and self.filters is None:
-                raise NameError('Beamforming weights or filters need to be computed first.')
-
-            # create window function
-            win = np.concatenate((np.zeros(self.zpf),
-                                  windows.hann(self.L),
-                                  np.zeros(self.zpb)))
-
-            # do real STFT of first signal
-            tfd_sig = stft.analysis(self.signals[0],
-                                self.L,
-                                self.hop,
-                                zp_back=self.zpb,
-                                zp_front=self.zpf,
-                                transform=np.fft.rfft,
-                                win=win) * np.conj(self.weights[0])
-            for i in range(1, self.M):
-                tfd_sig += stft.analysis(self.signals[i],
-                                     self.L,
-                                     self.hop,
-                                     zp_back=self.zpb,
-                                     zp_front=self.zpf,
-                                     transform=np.fft.rfft,
-                                     win=win) * np.conj(self.weights[i])
-
-            #  now reconstruct the signal
-            output = stft.synthesis(
-                tfd_sig,
-                self.L,
-                self.hop,
-                zp_back=self.zpb,
-                zp_front=self.zpf,
-                transform=np.fft.irfft)
+                raise NameError('Beamforming weights or filters need to be '
+                                'computed first.')
+
+            # create window functions
+            analysis_win = windows.hann(self.L)
+            # TODO: bad output when using this synthesis window, check
```

Comment:
I'm not entirely sure, I tried using a synthesis window that matched the analysis window, but the output sounded strange. Without the window, it sounded fine.

Reply:
Is a synthesis window not needed for beamforming?

Reply:
Well, not really since beamforming should be just a convolution, which is linear. In fact, even the analysis window is not needed. Only zero padding is necessary if a little care is taken to create causal filters only (i.e. delays are only positive) which can not always be guaranteed for some frequency domain beamformers.
I am wondering if the problem in our case is due to the use of synthesis window **and** zero-padding...

Reply:
Ok, what I'll do for now is remove this ugly comment, create a clean PR, and merge these other fixes.

Reply:
Sounds great.

Reply:
Huh, just realized there's a squash and merge option on GitHub itself! Won't do it manually next time 😛